### PR TITLE
Raises ArgumentError when nonrelative path used against child base_uri

### LINKED ISSUE #63: Leading slash on path ignores base_uri reference
For example,

```ruby
# DELETE
parent_ref = Firebase::Client.new('https://my-firebase-db.firebaseio.com/parent')
parent_ref.delete('/child')
```

... this will make a delete request to 'https://my-firebase-db.firebaseio.com/child', rather than 'https://my-firebase-db.firebaseio.com/parent/child'. This is because of the way the Addressable gem, used by httpclient, concatenates a base url with a path.

Is this what is intended? I found it quite surprising.

### PR DESCRIPTION
**Pull request by ptolemybarnes**
Closes #63

### CODE REVIEW

**Comment by ptolemybarnes** [comment]
CI is failing on Ruby 1.8.7. Can't seem to get that to run on my machine though :-/ Or, rather, bundler fails.

File: lib/firebase.rb [hunk 1]
```diff
@@ -53,11 +53,18 @@ def update(path, data, query={})
...
+      end
       Firebase::Response.new @request.request(verb, "#{path}.json", {
         :body             => (data && data.to_json),
         :query            => (@auth ? { :auth => @auth }.merge(query) : query),
         :follow_redirect  => true
       })
     end
+
+    def non_relative_path_against_child_base_url?(path)
+      path.lstrip[0] == '/' && URI(@request.base_url).path != "/"
```
Comment (vincentwoo) [inline lib/firebase.rb]: Why is lstrip required? Paths should not contain leading whitespace. If they do, we should let the code explode normally, I believe.

The path check is also brittle: `URI('https://asdf.io').path` yields empty string. I would prefer to just always throw an error if path starts with a slash.

Comment (ptolemybarnes) [inline lib/firebase.rb]: 👍  that simplifies things a lot.

Comment (ptolemybarnes) [inline lib/firebase.rb]: BTW a '/' is always added to the base_uri on line 15 so you wouldn't get an empty string at that point. But I see your point that it's more consistent to always throw an error when an absolute path is used, regardless of whether the base_uri is a root or child reference.

**Comment by vincentwoo** [comment]
Okay, this looks fine aside from the 1.8.7 error. I'll look into that when I can and try to get this merged. Thank you for the PR.

**Pull request merged**